oxidation/libparsec/crates/serialization_format/src/protocol.rs

```diff
@@ -0,0 +1,739 @@
+        let module_name = format_ident!("{}", &self.cmd);
+        match &self.spec {
+            GenCmdSpec::ReusedFromVersion { version } => {
+                let reused_version_module_name = format_ident!("v{}", version);
+                quote! {
+                    pub mod #module_name {
+                        pub use super::super::#reused_version_module_name::#module_name::*;
+                    }
+                }
+            }
+
+            GenCmdSpec::Original {
+                req,
+                reps,
+                nested_types,
+            } => {
+                let struct_req = quote_cmd_req_struct(req);
+                let variants_rep = quote_cmd_rep_variants(reps);
+                let nested_types = quote_cmd_nested_types(nested_types);
+                let known_rep_statuses: Vec<String> =
+                    reps.iter().map(|rep| rep.status.to_owned()).collect();
+
+                quote! {
+
+                    pub mod #module_name {
+                        use super::AnyCmdReq;
+                        use super::UnknownStatus;
+
+                        #(#nested_types)*
+
+                        #struct_req
+
+                        impl Req {
+                            pub fn dump(self) -> Result<Vec<u8>, ::rmp_serde::encode::Error> {
+                                AnyCmdReq::#variant_name(self).dump()
+                            }
+                        }
+
+                        // Can't derive Eq because some Rep have f64 field
+                        #[allow(clippy::derive_partial_eq_without_eq)]
+                        #[::serde_with::serde_as]
+                        #[derive(Debug, Clone, ::serde::Serialize, ::serde::Deserialize, PartialEq)]
+                        #[serde(tag = "status")]
+                        pub enum Rep {
+                            #(#variants_rep),*
+                        }
+
+                        impl Rep {
+                            pub fn dump(&self) -> Result<Vec<u8>, ::rmp_serde::encode::Error> {
+                                ::rmp_serde::to_vec_named(self)
+                            }
+
+                            pub fn load(buf: &[u8]) -> Result<Self, ::rmp_serde::decode::Error> {
+                                ::rmp_serde::from_slice::<Self>(buf)
+                                    .or_else(|err| {
+                                        // Due to how Serde handles variant discriminant, we cannot express unknown
+                                        // status as a default case in the main schema.
+                                        // Instead we have this additional deserialization attempt fallback.
+                                        let data = ::rmp_serde::from_slice::<UnknownStatus>(buf)?;
+
+                                        match data.status.as_str() {
+                                            #(#known_rep_statuses => Err(err),)*
```

Comment:
```suggestion
                                            #(#known_rep_statuses)|* => Err(err)
```

Could also be written that way, which is better :shrug:

Reply:
:+1: I don't know why, but I find your solution much more satisfying :smile: 

done.